wgengine/magicsock/magicsock.go

```diff
@@ -3014,7 +3014,7 @@ type RebindingUDPConn struct {
 	// meanwhile.
 	// pconn isn't really needed, but makes some of the code simpler
 	// to keep it in a type safe form.
```

Comment:
this comment is old

while you're here, document whether this atomic.Pointer can ever be nil and if so, what that means (it seems like it's never nil)

also, clarify that it's **not** a pointer _to_ the `(*RebindingUDPConn).pconn` value

might be worth adding a little test that just calls `setConnLocked` to two different PacketConn impl types lest anybody "optimize" this code in the future and revert this back to a syncs.AtomicValue.

Reply:
Now that I've ready the comment a bit more carefully, it says:
```
	// pconn isn't really needed, but makes some of the code simpler
	// to keep it in a type safe form.
```
But `pconnAtomic` is typesafe now, there used to be a TODO here that was removed in https://github.com/tailscale/tailscale/commit/a9f6cd41fdf82f55ed1f0e99291175fbd9a8d915#diff-7b44a473ce1a6ac00f78933d1f0651f64f7bd20de479235a324d108c3998cbc7L2982.

I still think we want `pconnAtomic` vs. `pconn` (to handle cases where the connection is rebound during a read/write), did my best to reword this.

Added a test.